plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -160,26 +153,28 @@ def project
 
     # helpful env vars, also useful for log tagging
     def set_env
-      env = (container[:env] ||= [])
-
-      static_env.each { |k, v| env << {name: k.to_s, value: v.to_s} }
-
-      # dynamic lookups for unknown things during deploy
-      {
-        POD_NAME: 'metadata.name',
-        POD_NAMESPACE: 'metadata.namespace',
-        POD_IP: 'status.podIP'
-      }.each do |k, v|
-        env << {
-         name: k.to_s,
-         valueFrom: {fieldRef: {fieldPath: v}}
-       }
-      end
+      containers.each do |cont|
+        env = (cont[:env] ||= [])
```

Comment:
added env is always the same ... we can prepare that outside the look and then only do add+unique inside ?